### PR TITLE
tests: Add unit tests for repeated_chars_iterator

### PR DESCRIPTION
**Pull request by RaduNichita**
Added two more unit tests to the suite:

* one to test the `std::next` function on a beman iterator
* one to test the `std::distance` function on a beman iterator

### CODE REVIEW

**Comment by steve-downey** [comment]
constexpr can involve different code paths in the implementation, so the tests should generally be both consteval and runtime. I think? 

I'm still working out understanding of constexpr testing and testing constexpr code.

**Review by neatudarius — APPROVED** [review]

**Comment by neatudarius** [comment]
I would also add tests for concepts like in https://github.com/beman-project/optional26/blob/main/src/beman/optional26/tests/detail/iterator.t.cpp

**Comment by neatudarius** [comment]
> constexpr can involve different code paths in the implementation, so the tests should generally be both consteval and runtime. I think?
> 
> I'm still working out understanding of constexpr testing and testing constexpr code.

https://github.com/beman-project/optional26/blob/main/src/beman/optional26/tests/detail/iterator.t.cpp#L18C9-L18C28

@steve-downey , is the same solution that we already have in this file in optional26? Is there any other "better" utility?

**Comment by steve-downey** [comment]
> > constexpr can involve different code paths in the implementation, so the tests should generally be both consteval and runtime. I think?
> > 
> > I'm still working out understanding of constexpr testing and testing constexpr code.
> 
> https://github.com/beman-project/optional26/blob/main/src/beman/optional26/tests/detail/iterator.t.cpp#L18C9-L18C28
> 
> @steve-downey , is the same solution that we already have in this file in optional26? Is there any other "better" utility? 

Looking at that, I'm not 100% sure that's actually consteval? That macro looks like it would work in a constexpr context, but it would also work at runtime? 
Also need to be sure that, for example, we could iterate an array in a consteval function using one of these? Or a constexpr vector, if the compiler will let us? 
And, hopefully, still produce a junit report so the dashboard is green, or the appropriate test is visibly failing? 

Failing to compile is a draconian error and we don't get any other test results.

File: tests/beman/iterator_interface/iterator_interface.test.cpp [hunk 1]
```diff
@@ -83,6 +137,7 @@ struct filtered_int_iterator
 };
 
 TEST(IteratorTest, TestFilteredIter) {
+
```
Comment (neatudarius) [inline tests/beman/iterator_interface/iterator_interface.test.cpp:140]:
```suggestion
```

**Review by neatudarius — APPROVED** [review]

**Comment by neatudarius** [comment]
Link this PR against #5

**Pull request merged**